geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/metadata/MetadataSerialization.scala

```diff
@@ -77,16 +87,19 @@ object MetadataSerialization extends MethodProfiling {
     case class StoragePersistence(
         featureType: Config,
         partitionScheme: PartitionSchemeConfig,
-        encoding: String,
-        leafStorage: Boolean
+        config: Map[String, String]
       )
+    case class StoragePersistenceV2(
+        featureType: Config, partitionScheme: PartitionSchemeConfig, encoding: String, leafStorage: Boolean)
     case class StoragePersistenceV1(featureType: Config, partitionScheme: PartitionSchemeConfig, encoding: String)
     case class PartitionSchemeConfig(scheme: String, options: Map[String, String])
 
     implicit val PartitionSchemeConfigConvert: ConfigConvert[PartitionSchemeConfig] =
       deriveConvert[PartitionSchemeConfig]
     implicit val StoragePersistenceV1Convert: ConfigConvert[StoragePersistenceV1] =
       deriveConvert[StoragePersistenceV1]
+    implicit val StoragePersistenceV2Convert: ConfigConvert[StoragePersistenceV2] =
```

Comment:
Since this PR is introducing a new metadata storage format version do we need to provide any notes about it in the upgrade guide?  Any utilities or notes about updating the metadata?

Reply:
no, this should be transparent to the end user.

Reply:
Any chance of issues with mixing and matching formats in the same directory?  E.g. if I wrote to the same feature type with multiple versions, would that work?

Reply:
no issue - if you write data with a newer version you won't be able to read it with an older version, which is (for better or worse) consistent with the rest of the project. i'm not sure of any way to change the format and still have older clients be able to read it, but i'm open to suggestions.

Reply:
Hmm... this change is additive right?  It only adds configuration to the metadata.  So it seems surprising that a reader would not be able to read data that's been written using a new, optional write-path configuration.  *blinks*

I don't have any particular suggestions at the minute.  I suppose I'd have to see the error which an older reader throws while trying to read metadata from this PR.

In some sense, I'd love to see 3.x GM FSDS tools which are "mix and match"....  (I suppose I'd argue that the gm cli tools give a kind of API and not being able to mix and match represents an API break that'd be great to capture with a major version bump.  I'm not firm on this stance; just worth discussing for a moment.)

Reply:
the 'encoding' and 'leaf storage' options were relocated, so that's why earlier versions can't read the new files. given our current data compatibility guarantees (https://www.geomesa.org/documentation/stable/user/upgrade.html#data-compatibility), we don't support mix-and-match between minor versions. we could change that and enforce things more strictly, but that's a larger discussion. i've added a note to the upgrade guide about the breaking change.